Extensions/shuffle+.js

```diff
@@ -1,148 +1,269 @@
+                        "button",
+                        {
+                            className: "checkbox" + (value ? "" : " disabled"),
+                            onClick: async () => {
+                                let state = !value;
+                                CONFIG[field] = state;
+                                setValue(state);
+                                await saveConfig();
+                                onclickFun();
+                            },
+                        },
+                        React.createElement(DisplayIcon, { icon: Spicetify.SVGIcons.check, size: 16 })
+                    )
+                )
+            );
+        }
 
-    /**
-     *
-     * @param {string} uri
-     * @returns {Promise<string[]>}
-     */
-    async function fetchListFromUri(uri) {
-        const uriObj = Spicetify.URI.fromString(uri);
+        function dropDownItem({ name, field, options, onclickFun = () => {} }) {
+            const [value, setValue] = useState(CONFIG[field]);
+            return React.createElement(
+                "div",
+                { className: "popup-row" },
+                React.createElement("label", { className: "col description" }, name),
+                React.createElement(
+                    "div",
+                    { className: "col action" },
+                    React.createElement(
+                        "select",
+                        {
+                            value,
+                            onChange: async (e) => {
+                                setValue(e.target.value);
+                                CONFIG[field] = e.target.value;
+                                await saveConfig();
+                                onclickFun();
+                            },
+                        },
+                        Object.keys(options).map((item) =>
+                            React.createElement(
+                                "option",
+                                {
+                                    value: item,
+                                },
+                                options[item]
+                            )
+                        )
+                    )
+                )
+            );
+        }
+
+        let settingsDOMContent = React.createElement(
+            "div",
+            null,
+            style,
+            React.createElement("div", { className: "popup-row" }, React.createElement("h3", { className: "div-title" }, "Artist Shuffle")),
+            React.createElement("div", { className: "popup-row" }, React.createElement("hr", { className: "divider" }, null)),
+            React.createElement(dropDownItem, {
+                name: "Shuffle mode Artist Page",
+                field: "artistMode",
+                options: {
+                    all: "All",
+                    album: "Albums",
+                    single: "Singles & EP",
+                    likedSongArtist: "Liked Songs of Artist",
```

Comment:
```suggestion
                    likedSongArtist: "Artist's Liked Songs",
```

Reply:
no

Reply:
Grammatically incorrect. Please do change it.